db/sqlinterfaces.c

```diff
@@ -2711,11 +2698,18 @@ static int bind_parameters(struct reqlogger *logger, sqlite3_stmt *stmt,
             break;
         case CLIENT_CSTR:
         case CLIENT_PSTR:
-        case CLIENT_PSTR2:
-            fix_cstr_len((char *)p.u.p, &p.len);
+        case CLIENT_PSTR2: {
+            /* Keeping with the R6 behavior, we need to 'silently' truncate
+             * the string on first '\0'.
+             */
+            size_t len = strlen((char *)p.u.p);
```

Comment:
Input may not be null terminated. `p.len = strnlen(p.u.p, p.len)`?

Reply:
good point